src/odemis/acq/stream/_projection.py

```diff
@@ -218,6 +224,428 @@ def _updateImage(self):
+    Its .image contains a DataArray 2D RGB (shape YXC), with metadata MD_PIXEL_SIZE and MD_POS.
+    The .background and .point VA are still on the stream passed in the __init__ as the same
+    background image and ebeam position can be used for multiple projections connected to the same stream.
+    """
+
+    def __init__(self, stream):
+        """
+        :param stream: (Stream) The stream the projection is connected to.
+        """
+        super(ARPolarimetryProjection, self).__init__(stream)
+
+        # Cached conversion of the raw 6 polarization images to:
+        #   *Stokes parameters in the detector plane (4 images)
+        #   *Stokes parameters in the sample plane (4 images)
+        #   *E-fields in polar and Cartesian coordinates (5 images)
+        #   *DOPs (degree of polarization) (4 images)
+        # All images in polar representation.
+        self._polarimetry_cache = {}  # dict tuple (float, float, str or None) -> DataArray
+        # represents (ebeam posX, ebeam posY, polarimetry pos)
+        # Same as above, but the raw (aka rectangular representation) images.
+        self._polarimetry_cache_raw = {}  # dict tuple (float, float, str or None) -> DataArray
+        # TODO check if possible to store only raw -> would need to call Rectangular2Polar each time
+        # a new polarimetry pos is requested...
+
+        self.polarimetry = self.stream.polarimetry  # make VA an attribute of the projection
+        self.polarimetry.subscribe(self._onPolarimetry)
+
+    def _project2Stokes(self, pos):
+        """
+        Return the polarimetry visualization of the images at the requested ebeam and polarimetry position.
+        :param pos: (float, float, string or None) position (must be part of the .stream._pos_polarimetry)
+        :returns: (DataArray) The polarimetry visualization projection.
+        """
+        # Note: Need a copy of the link to the dict. If self._polarimetry_cache is reset while
+        # still running this method, the dict might get new entries again, though it should be empty.
+        polarimetry_cache = self._polarimetry_cache
+        polarimetry_cache_raw = self._polarimetry_cache_raw
+        # Note: Method needs about 4sec to display the image for selecting a new ebeam position
+        if pos in polarimetry_cache:
+            stokes_data = polarimetry_cache[pos]
+        else:
+            ebeam_pos = pos[:2]
+            # Compute the polarimetry representation
+            data_raw = self.stream._pos_polarimetry[ebeam_pos]  # get the 6 images for requested ebeam pos
+            try:
+                # TODO get the raw/bg processed data from polar_cache, as now we do bg subtraction twice
+                data_raw_bg_corr = {}
+                # Correct image for background. It must match the polarization (defaulting to MD_POL_NONE).
+                for key in data_raw.keys():
+                    data_raw_bg_corr[key] = self._processBackground(data_raw[key],
```

Comment:
For such loop, use `.items()`. Also try to pick explicit names for the key and value (so that the code is self-documenting). So something like:
```
for pol, raw in data_raw.items():
   calibrated = self._processBackground(raw,  raw.metadata.get(model.MD_POL_MODE, model.MD_POL_NONE))
   data_raw_bg_corr[pol] = calibrated
```